### PR TITLE
Automate ssc differ

### PR DESCRIPTION
**Pull request by dguittet**
## Description

Automating the SSC Variable and SAM default differ to run in Github Actions

Requires an easy way to query and download the previous SAM version, this is done by adding the download paths to the Github Releases: such as for the 2021.12.02.r0.ssc.267 release here: https://github.com/NREL/SAM/releases/tag/2021.12.02.r0.ssc.267

Steps to Use:
1. Create a new Release in SAM repo and make it a "Pre-release". Add the download links
2. Trigger SAM's Github Actions by making a Push or a Pull Request
3. Download the Artefact version_diff.json from Github Actions and use in Release Notes

 Here's an example Github Action run: https://github.com/NREL/SAM/actions/runs/1809701751

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

If you have added a new compute module in a SSC pull request related to this one, be sure to check the [Process Requirements](https://github.com/NREL/SAM/wiki/Compute-modules-in-SAM).

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by sjanzou** [comment]
@dguittet, looking at the example https://github.com/NREL/SAM/actions/runs/1809701751 and the failing check, what actions should be taken by whoever is running the action? Should this be a pull request failure in the future?

**Comment by dguittet** [comment]
@sjanzou That example is successful. The latest failed due to differences in the test script of the defaults. I'm not sure why that is as this PR has no impact on the test files or those results.

**Comment by brtietz** [comment]
@dguittet I suspect these are due to underlying SSC changes. It seems reasonable that https://github.com/NREL/ssc/pull/743 would impact the stand alone battery LCOE.

**Comment by dguittet** [comment]
@brtietz That makes sense it's SSC changes. I added upload of the test_results_linux64.csv to Github Actions so we can use it in the test results update process.

**Comment by dguittet** [comment]
@brtietz It would be great to have a Windows test eventually! But it doesn't have to be right now unless you happen to already have it on your list.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -136,3 +139,26 @@ jobs:
...
       run: |
         python $SAMNTDIR/test/compare_lk_test_output.py ${SAMNTDIR}/test_results_win64.csv ${SAMNTDIR}/build/test_results_linux64.csv true
+
+    - name: Upload Artifacts
+      uses: actions/upload-artifact@v2
+      with:
+          name: Test Results Linux
+          path: ${{env.SAMNTDIR}}/build/test_results_linux64.csv
+
+    - name: Set up Python ${{ matrix.python-version }}
```
Comment (brtietz) [inline .github/workflows/ci.yml:149]: Given that the evaluate test results steps also Python, should this be moved earlier in the test file?

Comment (dguittet) [reply .github/workflows/ci.yml:149]: I'm happy to move it

**Review by brtietz — APPROVED** [review]
Nice work!

**Comment by sjanzou** [comment]
@dguittet , did these changes get merged into develop, too?

**Comment by dguittet** [comment]
@sjanzou Thanks for the reminder!

**Pull request merged**